core/evm.go

```diff
@@ -96,18 +106,31 @@ func newEVMBlockContext(header *types.Header, chain ChainContext, author *common
 	if header.ExcessBlobGas != nil {
 		blobBaseFee = eip4844.CalcBlobFee(*header.ExcessBlobGas)
 	}
+
+	// Durango enables the Shanghai upgrade of eth, which takes place after the
+	// Merge upgrade.
+	isDurango := params.GetExtra(chain.Config()).IsDurango(header.Time)
+	if isDurango {
+		random = new(common.Hash)
+		random.SetBytes(header.Difficulty.Bytes())
```

Comment:
it should not be dangerous since we will not serialize this value back.
it is used by the "random" opcode, so we keep it as "difficulty", which retains the previous semantics.
This way we can avoid changing the jump table compared with upstream, which seems better.
however, I do agree that it's not in correspondence with the block header.